cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -0,0 +1,166 @@
+from cg.io.controller import WriteFile, WriteStream
+from cg.meta.demultiplex.housekeeper_storage_functions import (
+    add_and_include_sample_sheet_path_to_housekeeper,
+)
+from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
+
+LOG = logging.getLogger(__name__)
+
+
+class SampleSheetAPI:
+    """Sample Sheet API class."""
+
+    def __init__(self, flow_cell_dir: str, hk_api: HousekeeperAPI, lims_api: LimsAPI) -> None:
+        self.flow_cell_runs_dir = Path(flow_cell_dir)
+        self.hk_api: HousekeeperAPI = hk_api
+        self.lims_api: LimsAPI = lims_api
+        self.dry_run: bool = False
+        self.force: bool = False
+        self.validator = SampleSheetValidator()
+
+    def set_dry_run(self, dry_run: bool) -> None:
+        """Set dry run."""
+        LOG.debug(f"Set dry run to {dry_run}")
+        self.dry_run = dry_run
+
+    def set_force(self, force: bool) -> None:
+        """Set force."""
+        LOG.debug(f"Set force to {force}")
+        self.force = force
+
+    def get_flow_cell(self, flow_cell_name: str, bcl_converter: str) -> FlowCellDirectoryData:
+        """
+        Return a flow cell given a path and the bcl converter.
+        Raises:
+            SampleSheetError: If the flow cell directory or the data it contains is not valid.
+        """
+        flow_cell_path: Path = Path(self.flow_cell_runs_dir, flow_cell_name)
+        if not flow_cell_path.exists():
+            message: str = f"Could not find flow cell {flow_cell_path}"
+            LOG.warning(message)
+            raise SampleSheetError(message)
+        try:
+            flow_cell = FlowCellDirectoryData(
+                flow_cell_path=flow_cell_path, bcl_converter=bcl_converter
+            )
+        except FlowCellError as error:
+            raise SampleSheetError from error
+        return flow_cell
+
+    def get_valid_sample_sheet_path(self, sample_sheet_path: Path) -> Path | None:
+        """Return the sample sheet path if it exists and if it passes validation."""
+        if sample_sheet_path and sample_sheet_path.exists():
+            try:
+                self.validator.validate_sample_sheet_from_file(sample_sheet_path)
+            except (SampleSheetError, ValidationError):
+                LOG.warning(f"Sample sheet {sample_sheet_path} was not valid")
+                return
+            return sample_sheet_path
+        else:
+            LOG.warning(f"Sample sheet with path {sample_sheet_path} does not exist")
+
+    def get_valid_sample_sheet_path_from_hk(self, flow_cell_id: str) -> Path | None:
+        """Return the sample sheet path from Housekeeper if is valid and exists."""
+        try:
+            sample_sheet_path: Path | None = self.hk_api.get_sample_sheet_path(flow_cell_id)
+        except HousekeeperFileMissingError:
+            LOG.warning(f"Sample sheet for flow cell {flow_cell_id} not found in Housekeeper")
+            return
```

Comment:
Does it make more sense to raise an exception here instead?

Reply:
This is going to be called through `samplesheet create`. At first instance, the command will look for the sample sheet in Housekeeper, if it does not exist it just needs to create a new one. That is the purpose of the command, so there is no need to stop the execution.

Reply:
changed this, see updated version